net/retriever/retriever.go

```diff
@@ -92,10 +93,16 @@ func (m *MessageRetriever) fetchPointers() {
 	// Wait for each goroutine to finish then process any remaining messages that needed
 	// to be processed last
 	wg.Wait()
-	for _, env := range m.messageQueue {
-		m.handleMessage(env, nil)
+
+DRAIN_LOOP:
+	for {
+		select {
+		case env := <-m.messageQueue:
+			m.handleMessage(env, nil)
+		default:
+			break DRAIN_LOOP
```

Comment:
@tyler-smith When is default called. I see this breaks out of the loop but I'm not sure what triggers it.